File: NurseryAngularApplication/NurseryAngularApplication/ClientApp/src/app/services/service-file/farmer.ts
export class Farmer implements IFarmer {

  id: number;
  nurseryId: number;
  firstName: string;
  middleName: string;
  lastName: string;
  dateTime: string;
  contactNumber: number;
  email: string;
  villageName: string;
  genderType: number;
  educationType: number;
  stateId: number;
  districtId: number;
  tehsilId: number;

  constructor(data?: IFarmer) {
    if (data) {
      for (var property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }
  }

}

export interface IFarmer {
  id: number;
  nurseryId: number;
  firstName: string;
  middleName: string;
  lastName: string;
  dateTime: string;
  contactNumber: number;
  email: string;
  villageName: string;
  genderType: number;
  educationType: number;
  stateId: number;
  districtId: number;
  tehsilId: number;
}

File: NurseryAngularApplication/NurseryAngularApplication/Model/Farmers/Farmer.cs
﻿using NurseryAngularApplication.Model.Employee;
using NurseryAngularApplication.Model.Nurserys;
using NurseryAngularApplication.Model.States;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace NurseryAngularApplication.Model.Farmers
{
    public class Farmer : CommonDataModel
    {
        [Key]
        public int Id { get; set; }

        public int? NurseryId { get; set; }
        public Nursery Nursery { get; set; }

        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }

        public DateTime? DateofBirth { get; set; }

        public string ContactNumber { get; set; }

        public string Email { get; set; }
        public string VillageName { get; set; }
        public GenderType GenderType { get; set; }

        public EducationType EducationType { get; set; }

        public int? StateId { get; set; }
        public State State { get; set; }

        public int? DistrictId { get; set; }
        public District District { get; set; }

        public int? TehsilId { get; set; }
        public Tehsil Tehsil { get; set; }
    }
}
